### PR TITLE
Fix Python version number in `cookiecutter --version` and test on Python 3.10

### LINKED ISSUE #1619: `cookiecutter --version` prints out the wrong Python version on 3.10
``` console
$ cookiecutter --version
Cookiecutter 1.7.3 from /usr/lib/python3.10/site-packages (Python 3.1)
```

It should end with `(Python 3.10)` not `(Python 3.1)`. This seems to be the problematic code:

https://github.com/cookiecutter/cookiecutter/blob/f821895a0cb536a907ac2937471d7d4ac36fec75/cookiecutter/cli.py#L26-L31

### PR DESCRIPTION
**Pull request by ozer550**
Closes #1619

- On Python 3.10 `cookiecutter --version` now prints Python 3.10 instead of Python 3.1
- Added test to verify this behavior 
- Added cross-platform Python 3.10 CI environments

### CODE REVIEW

**Review by audreyfeldroy — APPROVED** [review]
We reviewed this together over screenshare and I think this looks good, but I welcome others to review this.

**Comment by gotmax23** [comment]
I thought f-strings were not supported on Python 2.7 and 3.5, which this project purports it supports. Am I missing something?

**Comment by michaeljoseph** [comment]
@gotmax23 where did you find references to those legacy python versions?
We don't support them any longer (as you can see from the CI runs).

File: cookiecutter/cli.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 
 def version_msg():
     """Return the Cookiecutter version, location and Python powering it."""
-    python_version = sys.version[:3]
+    python_version = f'{sys.version_info.major}.{sys.version_info.minor}'
```
Comment (michaeljoseph) [inline cookiecutter/cli.py:28]: Any reason we wouldn't want to include the full version here?

**Comment by gotmax23** [comment]
> @gotmax23 where did you find references to those legacy python versions? We don't support them any longer (as you can see from the CI runs).

It's listed in one of the shields.io badges in the README. I guess that whatever data source it pulls from is outdated.

**Review by michaeljoseph — APPROVED** [review]
🙌

**Pull request merged**